pf4j-spring/src/main/java/org/pf4j/spring/ExtensionsInjector.java

```diff
@@ -31,17 +31,17 @@
 
     private static final Logger log = LoggerFactory.getLogger(ExtensionsInjector.class);
 
-    protected final PluginManager pluginManager;
+    protected final SpringPluginManager springPluginManager;
```

Comment:
Is this change really needed?

Reply:
From my assessment of the ExtensionsInjector class, it seems that the class is primarily used for registration of extensions to the application context of the SpringPluginManager only. Instead of passing in the SpringPluginManager as a boxed up parameter, I thought we should pass in as SpringPluginManager instead so that we could retrieve the application context, to make use of the 'getBeansOfType' method to retrieve all registered instances of the extension class. Another alternative is to use the 'getBean' method of the beanFactory parameter but that involves catching 'NoSuchBeanDefinitionException' and registering the extension in the handler.